Build runnable alfacase built with SCORE data

gabrielantao opened the pull request:
Configure some default values for configured case in order to have a runnable case. Functions to parse result data and extract to csv were added.

NOTE: The current case configured from example (in the tests folder) is running, but it generates dubious results, different of the SCORE results for pressure and temperature (see the Jira task for the plots).

PWPA-1995

Commit:
Add the fixed to get a first runnable case;

PWPA-1995

Commit:
Add output reader class;

PWPA-1995

Commit:
Add some default values for the alfacase

PWPA-1995

Commit:
Add a test for the SCORE export profiles;

PWPA-1995

carlosnewmar commented on src/alfasim_score/converter/alfacase/score_input_reader.py:
```diff
@@ -312,3 +313,35 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
             "gas_gravity": Scalar(fluid_data["gas_gravity"], FRACTION_UNIT),
         }
+
+    def read_output_curves(self) -> Dict[str, Array]:
```
Include a comment in both methods, that they are only use to "test", compare results quem presented in json.

gabrielantao replied:
Ok, added the comment

carlosnewmar commented on src/alfasim_score/converter/alfacase/score_input_reader.py:
```diff
@@ -312,3 +313,35 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
+
+    def read_output_curves(self) -> Dict[str, Array]:
+        """Get the results from the input file."""
+        result = self.input_content["operation"]["thermal_simulation"]["result"]["0"]
```
Here you are expecting that it exist, I'd rather use get, and check if it exist.

gabrielantao replied:
Indeed, I check now if I have result.

carlosnewmar commented on src/alfasim_score/converter/alfacase/base_operation.py:
```diff
@@ -123,7 +208,65 @@ def build_nodes(self) -> List[NodeDescription]:
+                temperatures=InitialTemperaturesDescription(
+                    position_input_type=TableInputType.length,
+                    table_length=TemperaturesContainerDescription(
+                        positions=Array([well_length.GetValue()], LENGTH_UNIT),
```
Here you could use the formation temperature.
Keep, the bottom position/temperature as it was included, however include the top position (0 m) and the top temperature as defined in environment.

gabrielantao replied:
Ok, sounds good, done.

gabrielantao replied:
With this change the case converges ok.

carlosnewmar commented on src/alfasim_score/converter/alfacase/base_operation.py:
```diff
@@ -123,7 +208,65 @@ def build_nodes(self) -> List[NodeDescription]:
+        well_length = self.get_position_in_well(self.score_input.read_general_data()["final_md"])
+        operation_data = self.score_input.read_operation_data()
+        initial_bottom_pressure = operation_data["flow_initial_pressure"].GetValue(PRESSURE_UNIT)
+        # the factor multiplied for the top pressure is arbitrary, just to give a initial value
```
:+1:

gabrielantao replied:
😄

carlosnewmar commented on src/alfasim_score/converter/alfacase/base_operation.py:
```diff
@@ -97,6 +172,16 @@ def build_nodes(self) -> List[NodeDescription]:
                     temperature=operation_data["flow_initial_temperature"],
                     pressure=operation_data["flow_initial_pressure"],
                     split_type=MassInflowSplitType.Pvt,
```
If you are using `split_type=MassInflowSplitType.Pvt,`, you don't need to define `volume_fraction`and `mass_fraction`.
`mass_fraction` is required when using: `MassInflowSplitType.ConstantMassFraction` and `volume_fraction`, when using: `MassInflowSplitType.ConstantVolumeFraction`

gabrielantao replied:
Alright. Removed.

Commit:
Use formation temperature at well start position;

Remove volume and mass fractions because of choosen split;
Improve comments of test methods in reader ;

PWPA-1995

carlosnewmar commented on src/alfasim_score/converter/alfacase/score_input_reader.py:
```diff
@@ -312,3 +313,45 @@ def read_operation_fluid_data(self) -> Dict[str, Any]:
+        Get the results from the input file.
+        This parsed output curves are used to check cases results.
+        """
+        if "result" in self.input_content["operation"]["thermal_simulation"]:
```
A more "fancie" approach could be:
```
if results := self.input_content["operation"]["thermal_simulation"].get("result"):
    result = results["0"]
```

gabrielantao replied:
Honestly, I think this syntax is terrible hahahhaha.
